app/code/Magento/CatalogGraphQl/Service/CheckImageCacheFileService.php

```diff
@@ -0,0 +1,123 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CatalogGraphQl\Service;
+
+use Magento\Catalog\Model\Product\Image;
+use Magento\Catalog\Model\Product\ImageFactory;
+use Magento\Catalog\Model\Product\Image\ParamsBuilder;
+use Magento\Framework\App\Filesystem\DirectoryList;
+use Magento\Framework\Exception\FileSystemException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\Filesystem;
+use Magento\Store\Model\StoreManagerInterface;
+
+/**
+ * Service check if image cache file exist
+ */
+class CheckImageCacheFileService
+{
+    /**
+     * @var Filesystem\Directory\WriteInterface
+     */
+    private $mediaDirectory;
+
+    /**
+     * @var ParamsBuilder
+     */
+    private $paramsBuilder;
+
+    /**
+     * @var ImageFactory
+     */
+    private $productImageFactory;
+
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
+     * @param ImageFactory $imageFactory
+     * @param Filesystem $filesystem
+     * @param ParamsBuilder $paramsBuilder
+     * @param StoreManagerInterface $storeManager
+     * @throws FileSystemException
+     */
+    public function __construct(
+        ImageFactory $imageFactory,
+        Filesystem $filesystem,
+        ParamsBuilder $paramsBuilder,
+        StoreManagerInterface $storeManager
+    ) {
+        $this->productImageFactory = $imageFactory;
+        $this->mediaDirectory = $filesystem->getDirectoryWrite(DirectoryList::MEDIA);
+        $this->paramsBuilder = $paramsBuilder;
+        $this->storeManager = $storeManager;
+    }
+
+    /**
+     * Save image cache file if is not exist
+     *
+     * @param string $imgUrl
+     * @param string $imagePath
+     * @param string $imageType
+     * @return string
+     * @throws NoSuchEntityException
+     */
+    public function execute(
+        string $imgUrl,
+        string $imagePath,
+        string $imageType
+    ) {
+        if ($imgUrl) {
```

Comment:
I would recommend you to use negative checks instead of layering your conditions like

```php
if {
    if {
        if {
...
```

It can be achieved in the following way 

```suggestion
        if (!$imgUrl) {
            return $imgUrl; // Or just return NULL if it's supposed to be returned in this case
        }
        // Then you may proceed here without adding layers to the the cyclomatic complexity
```